### DISCUSSION TITLE
Welcome to vqgan-clip-generator Discussions!

### DISCUSSION THREAD

**Discussion by rkhamilton** [rkhamilton/vqgan-clip-generator#3] [Announcements]
## Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
 
Feel free to reply below to introduce yourself.

**Comment by halr9000** [comment]
Hey @rkhamilton I just wanted to say thanks for this project. I've been going through all of the TTI projects past few weeks, and while I'm just an "experienced hacker", not a s/w dev, your project is one of the extremely few where I can tell the author:

- actually knows how to write software (and you apply principles such as DRY which others do not think to)
- cares about how Python should look and work, which makes it so easy to learn from
- knows how conda and pip work (UGGGHHH this one is a pain in other projects)
- you are thoughtful about how to get started, and likely tested that your instructions will work

I remember trying to get a project working the other day and grimacing when the instructions said "you need to use ipython because I was lazy". For some reason I kept beating my head against it, but I really should've stopped the instant I ran into py3 print syntax bugs in the dependencies after following the setup instructions...

Anyway, breath of fresh air, thanks again :D

**Reply by rkhamilton** [reply]
I appreciate the kind words. I'm a researcher in an unrelated field who has done written some simulations and statistical models, and some control code, and done some data science work. It's been my experience professionally, and also what I've seen in the AI art community, that there's not a ton of overlap between domain experts (in AI, artists, mathematicians) and people who understand and care about software design and "clean code." Software development is of course its own domain one can be an expert in. I'm a domain expert in another area, but I do care about clean code, and try to write code that adheres to good principles. I'm miles away from where a senior software developer would be, but I make an effort. It's nice to hear that someone appreciates it.

The image generation code really needs another refactoring pass to clean it up further, but I tried to make it more clear than what I started from. I was refactoring the original code as I was coming to understand it, so it's not really in an ideal form, but it makes more sense to me now than a flat script with repeated elements all over.

**Reply by halr9000** [reply]
I've not read a ton of the code yet, but working through the scripts and simply the fact that you got this down to the 3 functions is super easy to work with. I just forked the code and I /might/ get to the point that I send you some newer example scripts that will do argparsing for a CLI user. But do not wait on me, I am a very slow coder and tend to not finish ideas. :)
